service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -302,20 +302,21 @@ private JInvocation createRequest(ServiceIDL serviceIdl, JCodeModel jCodeModel,
                 if (endPoint.getRequestParamWithLimit() != null && requestParamWithLimit.equals(arg)) {
                     argRef = listElementVarName;
                 }
+                String paramName = Optional.ofNullable(parameter.getName()).orElse(argRef);
```

Comment:
.orElse(arg) instead of .orElse(argRef)
argRef in case of batching will contain the ListElement suffix.
Ex: If parameter name is id and we are doing request batching on this paramter, generated code will be
List<T> idList = RequestSplitter.split(id);
for (T idListElement: idList) {
   // at this point argRef will be idListElement and arg will be id.
}